src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -2428,4 +2435,42 @@ static OrganizationUserStatusType GetPriorActiveOrganizationUserStatusType(Organ
 
         return status;
     }
+
+    public async Task CreatePendingOrganization(Organization organization, string ownerEmail)
```

Comment:
I think we might be missing some code here (and/or in the `InitPendingOrganization` method). Look at the `SignUpAsync` methods. There are 3:

* `public async Task<Tuple<Organization, OrganizationUser>> SignUpAsync(OrganizationSignup signup, bool provider = false)` - creating an org in cloud

* `public async Task<Tuple<Organization, OrganizationUser>> SignUpAsync(OrganizationLicense license, User owner, string ownerKey, string collectionName, string publicKey, string privateKey)` - creating an org in self-hosted (not applicable here)

* `private async Task<Tuple<Organization, OrganizationUser>> SignUpAsync(Organization organization, Guid ownerId, string ownerKey, string collectionName, bool withPayment` - common code shared between both methods

The cloud method has a call to `PaymentService` which I assume we'll need to start billing this new org. It also calls `ValidateSignUpPoliciesAsync` which we'll want to do.

The private method saves the Default Collection, which you'll also need when you add that to the request object. It also creates the organization api key and does other important stuff.

Ideally you should re-use the private method if possible, and carefully check the public cloud method to see if you need to copy any logic from there.

I'm not sure when billing should start, my preference is when the org is created (not when the invite is accepted), but you may have to check that one with product.

Reply:
`The cloud method has a call to PaymentService which I assume we'll need to start billing this new org.`

That makes sense but now I have other questions.

`PurchaseOrganizationAsync(Organization org, PaymentMethodType paymentMethodType,
        string paymentToken, StaticStore.Plan plan, short additionalStorageGb,
        int additionalSeats, bool premiumAccessAddon, TaxInfo taxInfo)` 

How do I get some of these values like `PaymentMethodType` and `paymentToken`?

> The private method saves the Default Collection, which you'll also need when you add that to the request object.

Do you mean when the user accepts the invite it should also send a collection name?

Reply:
I'm not actually sure how billing works for this, now that you mention it. Because they're resellers, CS might manually set it up at their end. Please ask.

> Do you mean when the user accepts the invite it should also send a collection name?
Yes - see my review on the `clients` PR for more info.

Reply:
> Because they're resellers, CS might manually set it up at their end

That is exactly right, we don't have to worry about that here, the CS team does the billing manually.

Reply:
Thanks for checking!